slides/intro-03-what-makes-a-model.qmd

```diff
@@ -199,127 +209,114 @@ countdown::countdown(minutes = 5, id = "explore-tree-spec")
+  theme_bw(base_size = 18)
 ```
 :::
 
 ::: {.column width="40%"}
 :::
 :::
 
-## Logistic regression
+## Proportional hazards model
 
 ::: columns
-::: {.column width="60%"}
-```{r plot-logistic-reg}
+::: {.column width="50%"}
+```{r plot-ph}
 #| echo: false
 #| fig.width: 8
 #| fig.height: 7
 
-logistic_preds <- logistic_reg() %>% 
-  fit(class ~ A, data = dat) %>% 
-  augment(new_data = bin_midpoints) 
-
-plot_rates +
-  geom_line(aes(A, .pred_one, color = I(test_color)), linewidth = 2, alpha = 0.8, data = logistic_preds)
+p_ph <- fit(proportional_hazards(), event_time ~ intake_condition, data = cat_train) %>% 
+  augment(new_data = cat_new, eval_time = seq(0, 360, 1)) %>% 
+  mutate(intake_condition = factor(intake_condition, levels = c("normal", "fractious", "feral", "other", "under_age_or_weight", "ill_moderatete", "ill_mild"))) %>% 
+  tidyr::unnest(cols = .pred) %>% 
+  ggplot() +
+  geom_step(aes(.eval_time, .pred_survival, col = intake_condition)) +
+  theme_bw(base_size = 18) +
+  theme(legend.position = "bottom")
+p_ph
+# plot_rates +
+#   geom_line(aes(A, .pred_one, color = I(test_color)), linewidth = 2, alpha = 0.8, data = logistic_preds)
 ```
 :::
 
-::: {.column width="40%"}
+::: {.column width="50%"}
 :::
 :::
 
-## Logistic regression
+## Proportional hazards model
 
 ::: columns
-::: {.column width="60%"}
-```{r plot-logistic-reg-2}
+::: {.column width="50%"}
+```{r plot-ph-2}
 #| echo: false
 #| fig.width: 8
 #| fig.height: 7
 
-plot_rates +
-  geom_line(aes(A, .pred_one, col = I(test_color)), linewidth = 2, alpha = 0.8, data = logistic_preds)
+p_ph
 ```
 :::
 
-::: {.column width="40%"}
+::: {.column width="50%"}
 
--   Logit of outcome probability modeled as linear combination of predictors:
+-   Hazard modeled via a baseline hazard and a linear combination of predictors:
 
-$log(\frac{p}{1 - p}) = \beta_0 + \beta_1\cdot \text{A}$
+$\lambda(t | x_i) = \lambda_0(t) \cdot \exp (x_i^T \beta)$
 
--   Find a sigmoid line that separates the two classes
+-   The hazard is proportional over all time $t$, and thus also the probability of survival is proportional.
 
 :::
 :::
 
-## Decision trees
+## Decision tree
 
 ::: columns
 ::: {.column width="50%"}
 ```{r tree-fit}
 #| echo: false
-#| fig.width: 8
-#| fig.height: 7
 
-tree_fit <- decision_tree(mode = "classification") %>% 
-  fit(class ~ A, data = mutate(dat, class = forcats::fct_rev(class)))
-
-tree_preds <- augment(tree_fit, new_data = bin_midpoints)
+fit_dt <- decision_tree(cost_complexity = 0.01) %>%
+  set_mode("censored regression") %>% 
+  fit(event_time ~ intake_condition, data = cat_train) 
 ```
 
 ```{r plot-tree-fit}
 #| echo: false
-#| fig.width: 4
-#| fig.height: 3.5
+#| fig.width: 6
+#| fig.height: 5
 #| fig-align: center
 
-library(rpart.plot)
-tree_fit %>%
-  extract_fit_engine() %>%
-  rpart.plot(roundint = FALSE)
+library(partykit)
+fit_pec <- fit_dt %>% extract_fit_engine()
+fit_rpart <- fit_pec$rpart
+fit_rpart %>% 
+  as.party() %>% 
+  plot()
```

Comment:
About 2 weeks ago, I discovered the [`justmin` argument](https://stackoverflow.com/a/55049430/1078601) to stop the overlaid split categories. 

You might try 

```
plot(ep_args = list(justmin = 15 abbreviate = TRUE), gp = gpar(fontsize = 10))
```

and see if that helps.

Reply:
Also, I just love that we can make this plot.

Reply:
Me too! It's so nice to have the survival curves in the leaves!

Reply:
The `justmin` find is gold, thanks so much! 🎉 I've not reduced the font size to try to keep is somewhat readable from further away.